src/components/settings/EditOrganizationInformationsDialog.tsx

```diff
@@ -0,0 +1,323 @@
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import {
+  EditOrganizationInformationsDialogFragment,
+  Lago_Api_Error,
+  OrganizationInformationsFragmentDoc,
+  UpdateOrganizationInput,
+  useUpdateOrganizationInformationsMutation,
+} from '~/generated/graphql'
+import { theme } from '~/styles'
+import CountryCodes from '~/public/countryCode.json'
+
+const FILE_MAX_SIZE = 800000
+
+const countryData: { value: string; label: string }[] = Object.keys(CountryCodes).map(
+  (countryKey) => {
+    return {
+      value: countryKey,
+      // @ts-ignore
+      label: CountryCodes[countryKey],
+    }
+  }
+)
+
+gql`
+  fragment EditOrganizationInformationsDialog on Organization {
+    id
+    logoUrl
+    name
+    legalName
+    legalNumber
+    email
+    addressLine1
+    addressLine2
+    zipcode
+    city
+    state
+    country
+  }
+
+  mutation updateOrganizationInformations($input: UpdateOrganizationInput!) {
+    updateOrganization(input: $input) {
+      ...OrganizationInformations
+      ...EditOrganizationInformationsDialog
+    }
+  }
+  ${OrganizationInformationsFragmentDoc}
+`
+
+export interface EditOrganizationInformationsDialogRef extends DialogRef {}
+
+interface EditOrganizationInformationsProps {
+  organization: EditOrganizationInformationsDialogFragment
+}
+
+export const EditOrganizationInformationsDialog = forwardRef<
+  DialogRef,
+  EditOrganizationInformationsProps
+>(({ organization }: EditOrganizationInformationsProps, ref) => {
+  const { translate } = useInternationalization()
+  const mounted = useRef(false)
+  const [updateOrganizationInformations] = useUpdateOrganizationInformationsMutation({
+    context: { silentErrorCodes: [Lago_Api_Error.UnprocessableEntity] },
+  })
+  const [logo, setLogo] = useState<string>()
+  const [logoUploadError, setLogoUploadError] = useState(false)
+  const hiddenFileInputRef = useRef<HTMLInputElement>(null)
+  const formikProps = useFormik<UpdateOrganizationInput>({
+    initialValues: {
+      legalName: organization?.legalName || '',
+      legalNumber: organization?.legalNumber || '',
+      email: organization?.email || '',
+      addressLine1: organization?.addressLine1 || '',
+      addressLine2: organization?.addressLine2 || '',
+      zipcode: organization?.zipcode || '',
+      city: organization?.city || '',
+      state: organization?.state || '',
+      country: organization?.country || undefined,
+    },
+    validationSchema: object().shape({
+      email: string().email('text_620bc4d4269a55014d493fc3'),
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async ({ ...values }, formikBag) => {
+      await updateOrganizationInformations({
+        variables: {
+          input: {
+            logo,
+            ...values,
+          },
+        },
+        refetchQueries: ['getOrganizationInformations'],
+      })
+      ;(ref as unknown as RefObject<DialogRef>)?.current?.closeDialog()
```

Comment:
Ok so this part is a bit tricky. You should only use it when really needed, ie when you have to actually do something in the submit that implies that we shouldn't close the dialog.
Here you're closing it no matter what, so on the onClick you could just go with : 

```tsx
onClick={async () => {
   await formikProps.submitForm()
   closeDialog()
   if (mounted.current) {
     formikProps.resetForm()
   }
}}
```

Reply:
Ok understood, fixed